sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -160,7 +160,8 @@ statement
         UNSET TBLPROPERTIES (IF EXISTS)? tablePropertyList             #unsetTableProperties
     | ALTER TABLE table=multipartIdentifier
         (ALTER | CHANGE) COLUMN? column=multipartIdentifier
-        (TYPE dataType)? (COMMENT comment=STRING)? colPosition?        #alterTableColumn
+        (TYPE dataType (NOT NULL)?)?
```

Comment:
It doesn't make sense to update nullability along. The design of the existing `UpdateColumnType` class also follows this.

Reply:
this syntax  seems not to be ANSI standard
the  dedicated clauses might be
```sql
ALTER [ COLUMN ] column_name { SET | DROP } NOT NULL
```